### ISSUE TITLE
Directives declared on extensions are not applied to the actual type.

### ISSUE THREAD

**Issue by michaelstaib** [ChilliCream/hotchocolate#656]
Not 100% sure if I'm reading the code right, but it appears I can't apply directives to extended types during stitching.

I have a directive `@validate` defined and registered with the schema. It's defined with `DirectiveLocation.Object`.

I'm stitching together a number of schemas and some extensions (all from `.graphql` files). I can verify that this all works if I apply the directive to a type defined in the `extensions.graphql`. However if I apply it to a type extension I get the error
> The directive `validate` was not specified in this schema and cannot be used.

For reference the key part of the `extensions.graphql` looks like

```GraphQL
extend type Query @validate {
  organization(organizationKey: String!): Organization!
}

type Mutation @validate {
    echo(value: String!): String!
}
```

In the above sample only the directive on `Query` causes the exception, `Mutation` is fine.
Looking at the source for `AddSchemaExtensionRewriter.AddDirectives` and `MergeContext` it appears the directives dictionary is only uses the schema document and not the extensions document.

**Comment by michaelstaib** [comment]
So, I verified that but it turns out that we do the lookup correctly since first we integrate new types into the schema and then apply the extension types:

```csharp
DocumentNode current = schema;

            if (newTypes.Count > 0 || newDirectives.Count > 0)
            {
                current = RemoveDirectives(current,
                    newDirectives.Select(t => t.Name.Value));
                current = RemoveTypes(current,
                    newTypes.Select(t => t.Name.Value));

                var definitions = schema.Definitions.ToList();
                definitions.AddRange(newTypes);
                definitions.AddRange(newDirectives);
                current = current.WithDefinitions(definitions);
            }
```

**Comment by michaelstaib** [comment]
But as it turns out we are not using that new schema consequently :) OK that is an easy fix then....

**Comment by michaelstaib** [comment]
Ok this one is now fixed with #665. I will make rc.5 which is also now the last one before we release this version. Would be good if you could verify that your issue is fixed with this one.
